ipaplatform/suse/tasks.py

```diff
@@ -42,5 +43,158 @@ def set_nisdomain(self, nisdomain):
+
+    def modify_nsswitch_pam_stack(self, sssd, mkhomedir, statestore,
+                                  sudo=True):
+        # pylint: disable=ipa-forbidden-import
+        from ipalib import sysrestore  # FixMe: break import cycle
+        # pylint: enable=ipa-forbidden-import
+        fstore = sysrestore.FileStore(paths.IPA_CLIENT_SYSRESTORE)
+        logger.debug('Enabling SSSD in nsswitch')
+        self.configure_nsswitch_database(fstore, 'group', ['sss'],
+                                         default_value=['compat'])
+        self.configure_nsswitch_database(fstore, 'passwd', ['sss'],
+                                         default_value=['compat'])
+        self.configure_nsswitch_database(fstore, 'shadow', ['sss'],
+                                         default_value=['compat'])
+        self.configure_nsswitch_database(fstore, 'netgroup', ['files','sss'],
+                                         preserve=False,
+                                         default_value=['files','nis'])
+        self.configure_nsswitch_database(fstore, 'automount', ['files','sss'],
+                                         preserve=False,
+                                         default_value=['files','nis'])
+        if sudo:
+            self.enable_sssd_sudo(fstore)
+        logger.debug('Enabling sss in PAM')
+        try:
+            ipautil.run([paths.PAM_CONFIG, '--add', '--sss'])
+            if mkhomedir:
+                logger.debug('Enabling mkhomedir in PAM')
+                try:
+                    ipautil.run([paths.PAM_CONFIG, '--add', '--mkhomedir',
+                                 '--mkhomedir-umask=0077'])
+                except ipautil.CalledProcessError:
+                    logger.debug('Failed to configure PAM mkhomedir')
+                    return False
+        except ipautil.CalledProcessError:
+            logger.debug('Failed to configure PAM to use SSSD')
+            return False
+        return True
+
+    def restore_pre_ipa_client_configuration(self, fstore, statestore,
+                                             was_sssd_installed,
+                                             was_sssd_configured):
+        if fstore.has_file(paths.NSSWITCH_CONF):
+            logger.debug('Restoring nsswitch from fstore')
+            fstore.restore_file(paths.NSSWITCH_CONF)
+        else:
+            logger.info('nsswitch not restored')
+            return False
+        try:
+            logger.debug('Removing sssd from PAM')
+            ipautil.run([paths.PAM_CONFIG, '--delete', '--mkhomedir'])
+            ipautil.run([paths.PAM_CONFIG, '--delete', '--sss'])
+            logger.debug('Removing sssd from PAM successed')
+        except ipautil.CalledProcessError:
+            logger.debug('Faled to remove sssd from PAM')
+            return False
+        return True
+
+    def disable_ldap_automount(self, fstore, statestore):
+        # SUSE does not use authconfig or authselect
+        return True
```

Comment:
I'm not 100% sure on this one. We still twiddle with nsswitch.conf in the base module. The debian module calls the base method.

Reply:
I believe I addressed this issue in the latest commit.